lib/cluster/index.js

```diff
@@ -453,15 +454,28 @@ Cluster.prototype.sendCommand = function (command, stream, node) {
         if (!random) {
           if (typeof targetSlot === 'number' && _this.slots[targetSlot]) {
             var nodeKeys = _this.slots[targetSlot];
-            var key;
-            if (to === 'all') {
-              key = utils.sample(nodeKeys);
-            } else if (to === 'slave' && nodeKeys.length > 1) {
-              key = utils.sample(nodeKeys, 1);
+            if (typeof to === 'function') {
+              var nodes =
+                  nodeKeys
+                  .slice(1)
+                  .map(function(key) {
+                    return _this.connectionPool.nodes.all[key];
+                  });
+              redis = to(nodes, command);
+              if (redis.constructor === Array) {
```

Comment:
`if (Array.isArray(redis))` may be better IMO, and we should check whether `redis` is null/undefined and default to the master node.